src/tclint/commands/checks.py

```diff
@@ -153,13 +153,24 @@ def check_arg_spec(
         raise CommandArgError(f"{msg}, expected one of {', '.join(subcommands.keys())}")
 
     switches = arg_spec["switches"]
+    positionals = arg_spec["positionals"]
     args_allowed = set(switches)
     args_required = {switch for switch in switches if switches[switch]["required"]}
     positional_args = []
 
     args = list(args)
     while len(args) > 0:
         arg = args.pop(0)
+        contents = arg.contents
+
+        ispos = False
+        for pos in positionals:
+            if pos["name"] == contents:
```

Comment:
Thanks for opening this PR! I'd prefer this [patch](https://github.com/nmoroze/tclint/issues/108#issuecomment-3288796111) as a solution, since the assumption here that an argument is a positional if its name in the spec matches the contents is not actually valid.

(What I'm doing for the builtin arguments right now is a bit of a hack due to the tclint command schema not playing well with how Tcl actually evaluates their "switches", noted in this [commit](https://github.com/nmoroze/tclint/commit/f3661b9e23682d627e2f3e2f822e15c278690062)).